src/types/config.ts

```diff
@@ -220,3 +221,578 @@ export type DataRetentionPolicy = {
+    ListenAddress: string;
+};
+
+export type ExperimentalSettings = {
+    ClientSideCertEnable: boolean;
+    ClientSideCertCheck: string;
+    EnableClickToReply: boolean;
+    LinkMetadataTimeoutMilliseconds: number;
+    RestrictSystemAdmin: boolean;
+    UseNewSAMLLibrary: boolean;
+};
+
+export type AnalyticsSettings = {
+    MaxUsersForStatistics: number;
+};
+
+export type ElasticsearchSettings = {
+    ConnectionUrl: string;
+    Username: string;
+    Password: string;
+    EnableIndexing: boolean;
+    EnableSearching: boolean;
+    EnableAutocomplete: boolean;
+    Sniff: boolean;
+    PostIndexReplicas: number;
+    PostIndexShards: number;
+    ChannelIndexReplicas: number;
+    ChannelIndexShards: number;
+    UserIndexReplicas: number;
+    UserIndexShards: number;
+    AggregatePostsAfterDays: number;
+    PostsAggregatorJobStartTime: string;
+    IndexPrefix: string;
+    LiveIndexingBatchSize: number;
+    BulkIndexingTimeWindowSeconds: number;
+    RequestTimeoutSeconds: number;
+    SkipTLSVerification: boolean;
+    Trace: string;
+};
+
+export type BleveSettings = {
+    IndexDir: string;
+    EnableIndexing: boolean;
+    EnableSearching: boolean;
+    EnableAutocomplete: boolean;
+    BulkIndexingTimeWindowSeconds: number;
+};
+
+export type DataRetentionSettings = {
+    EnableMessageDeletion: boolean;
+    EnableFileDeletion: boolean;
+    MessageRetentionDays: number;
+    FileRetentionDays: number;
+    DeletionJobStartTime: string;
+};
+
+export type MessageExportSettings = {
+    EnableExport: boolean;
+    ExportFormat: string;
+    DailyRunTime: string;
+    ExportFromTimestamp: number;
+    BatchSize: number;
+    GlobalRelaySettings: {
+        CustomerType: string;
+        SmtpUsername: string;
+        SmtpPassword: string;
+        EmailAddress: string;
+    };
+};
+
+export type JobSettings = {
+    RunJobs: boolean;
+    RunScheduler: boolean;
+};
+
+export type PluginSettings = {
+    Enable: boolean;
+    EnableUploads: boolean;
+    AllowInsecureDownloadUrl: boolean;
+    EnableHealthCheck: boolean;
+    Directory: string;
+    ClientDirectory: string;
+    Plugins: Dictionary<any>;
+    PluginStates: Dictionary<{Enable: boolean}>;
+    EnableMarketplace: boolean;
+    EnableRemoteMarketplace: boolean;
+    AutomaticPrepackagedPlugins: boolean;
+    RequirePluginSignature: boolean;
+    MarketplaceUrl: string;
+    SignaturePublicKeyFiles: string[];
+};
+
+export type DisplaySettings = {
+    CustomUrlSchemes: string[];
+    ExperimentalTimezone: boolean;
+};
+
+export type GuestAccountsSettings = {
+    Enable: boolean;
+    AllowEmailAccounts: boolean;
+    EnforceMultifactorAuthentication: boolean;
+    RestrictCreationToDomains: string;
+};
+
+export type ImageProxySettings = {
+    Enable: boolean;
+    ImageProxyType: string;
+    RemoteImageProxyURL: string;
+    RemoteImageProxyOptions: string;
+};
+
+export type AdminConfig = {
+    ServiceSettings?: ServiceSettings;
```

Comment:
These can all be non-optional if we're going by the server's definition of the type

Reply:
I think they might be optional because of the reducer, which I didn't end up adding the type for anyway! Lemme have a second look.